python/cudf/cudf/utils/ioutils.py

```diff
@@ -1077,10 +1082,20 @@ def get_filepath_or_buffer(
             # Doing this as `read_json` accepts a json string
             # path_or_data need not be a filepath like string
             if os.path.exists(paths[0]):
-                path_or_data = paths[0]
+                if allow_multiple_files and len(paths) > 1:
+                    path_or_data = paths
+                else:
+                    path_or_data = paths[0]
         else:
-            with fs.open(paths[0]) as f:
-                path_or_data = BytesIO(f.read())
+            if allow_multiple_files and len(paths) > 1:
+                buffers = []
+                for fpath in paths:
+                    with fs.open(fpath) as f:
+                        buffers.append(BytesIO(f.read()))
+                path_or_data = buffers
```

Comment:
Can trim this down to a list comprehension:
```suggestion
                path_or_data = [BytesIO(fs.open(fpath).read()) for fpath in paths]
```